framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -942,10 +942,18 @@ public AnnotatedTypeMirror getAnnotatedType(Tree tree) {
             return treeCache.get(tree).deepCopy();
         }
 
-        AnnotatedTypeMirror type;
         if (TreeUtils.isClassTree(tree)) {
-            type = fromClass((ClassTree)tree);
-        } else if (tree.getKind() == Tree.Kind.METHOD ||
+            ClassTree classTree = (ClassTree) tree;
+            AnnotatedTypeMirror type = getAnnotatedTypeOfClassTree(classTree);
```

Comment:
You could move the handling of ClassTrees before the existing check for the cache, move updating of the cache into `getAnnotatedTypeOfClassTree`, and still ensure that postprocess is called if needed.